addons/purchase/models/purchase_order_line.py

```diff
@@ -226,7 +226,7 @@ def write(self, values):
     @api.ondelete(at_uninstall=False)
     def _unlink_except_purchase_or_done(self):
         for line in self:
-            if line.order_id.state in ['purchase', 'done']:
+            if line.order_id.state in ['purchase', 'done'] and line.display_type not in ['line_note', 'line_section']:
```

Comment:
It's true that in the case of PO lines, `and not line.display_type` would have worked, but:
> Explicit is better than implicit.